libconnman-qt/networksession.cpp

```diff
@@ -24,86 +39,86 @@ NetworkSession::~NetworkSession()
 
 void NetworkSession::createSession()
 {
-    if (m_path.isEmpty())
+    if (d_ptr->m_path.isEmpty())
         return;
 
-    delete m_sessionAgent;
-    m_sessionAgent = new SessionAgent(m_path ,this);
-    connect(m_sessionAgent,SIGNAL(settingsUpdated(QVariantMap)),
-            this,SLOT(sessionSettingsUpdated(QVariantMap)));
+    delete d_ptr->m_sessionAgent;
+    d_ptr->m_sessionAgent = new SessionAgent(d_ptr->m_path ,this);
```

Comment:
Same as old code, but... here we have d_ptr->m_sessionAgent switch over. I do not see it getting cleared/deleted anywhere. Is it getting leaked?

Reply:
Shouldn't leak. 'this' on creating QObjects is usually parent which owns the object and it will be automatically deleted.